server/src/test/java/org/elasticsearch/search/geo/GeoBoundsGenericWriteableTests.java

```diff
@@ -0,0 +1,120 @@
+        double right = geoBoundingBox.right() - width / 4;
+        return new GeoBoundingBox(new GeoPoint(top, left), new GeoPoint(bottom, right));
+    }
+
+    @Override
+    protected GenericNamedWriteable copyInstance(GenericNamedWriteable original, TransportVersion version) throws IOException {
+        return copyInstance(original, version, version);
+    }
+
+    /**
+     * Read and write with different serialization versions
+     */
+    protected GenericNamedWriteable copyInstance(
+        GenericNamedWriteable original,
+        TransportVersion writeVersion,
+        TransportVersion readVersion
+    ) throws IOException {
+        try (BytesStreamOutput output = new BytesStreamOutput()) {
+            output.setTransportVersion(writeVersion);
+            output.writeGenericValue(original);
+            try (StreamInput in = new NamedWriteableAwareStreamInput(output.bytes().streamInput(), getNamedWriteableRegistry())) {
+                in.setTransportVersion(readVersion);
+                return readGenericValue(in);
+            }
+        }
+    }
+
+    private GenericNamedWriteable readGenericValue(StreamInput in) throws IOException {
+        Object obj = in.readGenericValue();
+        if (obj instanceof GenericNamedWriteable result) {
+            return result;
+        }
+        throw new IllegalStateException("Read result of wrong type: " + obj.getClass().getSimpleName());
+    }
+
+    @Override
+    protected NamedWriteableRegistry getNamedWriteableRegistry() {
+        return registry;
+    }
+
+    @Override
+    protected Class<GenericNamedWriteable> categoryClass() {
+        return GenericNamedWriteable.class;
+    }
+
+    public void testSerializationFailsWithOlderVersion() throws IOException {
+        TransportVersion valid = TransportVersion.V_8_500_070;
+        TransportVersion older = TransportVersion.V_8_500_069;
+        for (int runs = 0; runs < NUMBER_OF_TEST_RUNS; runs++) {
+            GenericNamedWriteable testInstance = createTestInstance();
+
+            // Test reading and writing with valid versions
+            GenericNamedWriteable deserializedInstance = copyInstance(testInstance, valid, valid);
+            assertEqualInstances(testInstance, deserializedInstance);
+
+            // Test writing fails with older version
+            try {
+                copyInstance(testInstance, older, valid);
+                fail("Expected exception to be thrown when writing with older TransportVersion");
+            } catch (Throwable e) {
+                assertThat(
+                    "Writing with older TransportVersion",
+                    e.getMessage(),
+                    containsString("[GeoBoundingBox] requires minimal transport version")
+                );
+            }
+
+            // Test reading succeeds with older version (we do not cripple the reader)
+            deserializedInstance = copyInstance(testInstance, valid, older);
```

Comment:
I think this doesn't really make sense, we are always sure that reading and writing always use the same transport version, and the message we're creating isn't actually something an older node would be able to create.

Reply:
OK. I've simplified this failure test to only test the failure case. The first part with (valid,valid) was already tested in another test `testSerialization`, and the last part is unnecessary: